### ISSUE TITLE
Feature request: Patterned textures for shapes

### ISSUE THREAD

**Issue by merijn** [diagrams/diagrams-lib#321]
Having the ability to have shapes textured with patterns of other shapes would considerably increase the flexibility of diagrams and help users like Chart to implement, for example, textured bar charts.

**Comment by fryguybob** [comment]
Yes!  This is a feature we have thought about in the past that would be nice to have.  Last time I thought about the problem we didn't have any CSG functionality.  Some backends have this notion natively, others would need other support.  I think we have all the tools now and we need to think about the design.
